### PR TITLE
Fixes + improvements

### PR DESCRIPTION
**Pull request by GinjaNinja32**
- Omni filters can filter water
- Air alarms can set scrubbers to scrub water; scrubbers can scrub water
- Inflatables correctly block air/water
- `ChangeTurf()` calls `Destroy()` for ocean turf cleanup etc. Simulated wall `Destroy()` changed to not cause an infinite loop.
- `/obj/structure/Destroy()` and `/obj/effect/Destroy()` now wait until they're in nullspace to update atmos

### CODE REVIEW

**Comment by Zuhayr** [comment]
:+1:

File: code/game/objects/structures/inflatable.dm [hunk 1]
```diff
@@ -163,6 +166,9 @@
 		return !opacity
 	return !density
 
+/obj/structure/inflatable/door/CanAtmosPass(turf/T)
```
Comment (Zuhayr) [inline code/game/objects/structures/inflatable.dm]: I would have sworn this was already defined somewhere, but I may be thinking of CanPass().

Comment (GinjaNinja32) [inline code/game/objects/structures/inflatable.dm]: I kept having water flowing straight through my inflatables when I tried to make airlocks, so almost certainly not (or broken if it is).

**Comment by GinjaNinja32** [comment]
Now also resolves #8.

**Comment by SkowronX** [comment]
Good to go?

**Comment by Zuhayr** [comment]
:+1:

**Pull request merged**